packages/snaps-execution-environments/src/common/endowments/hardenedEndowments.ava.test.ts

```diff
@@ -0,0 +1,290 @@
+// eslint-disable-next-line import/no-unassigned-import
+import 'ses';
+// @ts-expect-error Walker has no types yet.
+import Walker from '@lavamoat/walker';
+import test from 'ava';
+// FinalizationRegistry will fix type errors in tests related to network endowment.
+// eslint-disable-next-line import/no-extraneous-dependencies, @typescript-eslint/no-unused-vars
+import FinalizationRegistry from 'globals';
+
+import buildCommonEndowments from './commonEndowmentFactory';
+import Crypto from './crypto';
+import date from './date';
+import interval from './interval';
+import math from './math';
+import network from './network';
+import timeout from './timeout';
+
+const globalThis = global;
+
+// Note: harden is only defined after calling lockdown
+lockdown({
+  domainTaming: 'unsafe',
+  errorTaming: 'unsafe',
+  stackFiltering: 'verbose',
+});
+
+/**
+ * Object walker test utility function.
+ *
+ * @param start - Subject to be tested.
+ * @param end - Target object.
+ * @returns True if target object is found, false otherwise.
+ */
+function walkAndSearch(start: unknown, end: unknown) {
+  let result = false;
+  const walker = new Walker(
+    (val: unknown) => {
+      return (result = result || val === end);
```

Comment:
This is how Gal designed it to be used. It's taken from a demo: https://github.com/weizman/lavamoat-walker-demo-1/blob/main/index.js
It requires the assignment to work.

Reply:
I think it's not possible or I don't have a proper idea to rewrite this, since this is how it works now and it's the external library which I cannot change now. Only if we maybe request a change from the maintainers in the future. But, the Walker library is doing these things in a different way and is not very specific to our use case. 

It requires a callback which is passed here in the constructor. I do not see the other way to return `true` or `false` properly as the end result which I need for the test assertion.

Reply:
'''
result = end === value;
return result;
'''

Should work too

Reply:
Is this what you asked for? @Mrtenz

Reply:
Yeah, that seems more understandable.

Reply:
Yes it works, I've updated with the suggestion. Thanks!